pkg/scheduler/internal/cache/cache.go

```diff
@@ -708,6 +710,24 @@ func (cache *cacheImpl) addNodeImageStates(node *v1.Node, nodeInfo *framework.No
 	nodeInfo.ImageStates = newSum
 }
 
+// updateNodeImageStates update node record from image entries having the node
+// in imageStates cache when receive node add/update/delete events.
+func (cache *cacheImpl) updateNodeImageStates(newNode *v1.Node) {
+	for existNode := cache.headNode; existNode != nil; existNode = existNode.next {
+		node := existNode.info.Node()
+		if node == nil || node.Name == newNode.Name {
+			continue
+		}
+		newSum := make(map[string]*framework.ImageStateSummary)
+		for image := range cache.imageStates {
+			if cache.imageStates[image].nodes.Has(node.Name) {
```

Comment:
why not just loop the map by k/v instead? so here you can deal with the value directly.

Reply:
thanks for your suggestion